go-controller/pkg/ovn/egressgw.go

```diff
@@ -39,6 +41,114 @@ type ovnRoute struct {
 	shouldExist bool
 }
 
+type externalRouteInfo struct {
+	sync.RWMutex
+	// podExternalRoutes is a cache keeping the LR routes added to the GRs when
+	// the k8s.ovn.org/routing-external-gws annotation is used. The first map key
+	// is the podIP, the second the GW and the third the GR
+	routes map[string]map[string]string
+}
+
+func (oc *Controller) ensureRouteInfoLocked(podName ktypes.NamespacedName, readOnly bool) (*externalRouteInfo, func(), error) {
+	oc.egressGWCacheMutex.Lock()
+	routeInfo := oc.egressGWCache[podName]
+	routeInfoExisted := false
+	if routeInfo == nil {
+		routeInfo = &externalRouteInfo{
+			routes: make(map[string]map[string]string),
+		}
+		// we are creating routeInfo and going to set it in podExternalRoutes map
```

Comment:
```suggestion
		// we are creating routeInfo and going to set it in routes map
```

Reply:
I can rename it if that is more clear

Reply:
yea let's rename it to `podExternalRoutes`, so it looks like we are literally picking that off nsInfo into externalRouteInfo